test/ehttpc_tests.erl

```diff
@@ -280,108 +280,89 @@ health_check_abnormal_test_() ->
     Unreachable = "8.8.8.8",
     Unknown = "unknown-host0",
     [
-        {timeout, 20, fun() ->
-            ?WITH(
-                #{
-                    port => Port,
-                    name => ?FUNCTION_NAME,
-                    delay => 0
-                },
-                pool_opts(Port, true),
-                begin
-                    Worker = ehttpc_pool:pick_worker(?POOL),
-                    %% 1. Do health_check with very short timeout,
-                    %% The check failed but ehttpc is still trying to establish
-                    %% the connection.
-                    ?assertEqual(
-                        {error, connect_timeout},
-                        ehttpc:health_check(Worker, 0)
-                    ),
-                    %% 2. do health_check again should be OK.
-                    ?assertEqual(ok, ehttpc:health_check(Worker, timer:seconds(2))),
-                    {ok, _} = ?block_until(
-                        #{?snk_kind := health_check_when_gun_client_not_ready},
-                        1000,
-                        infinity
-                    )
-                end
-            )
-        end},
-        {timeout, 20, fun() ->
-            ?WITH(
-                #{
-                    port => Port,
-                    name => ?FUNCTION_NAME,
-                    delay => 0
-                },
-                pool_opts(Port, true),
-                begin
-                    Worker = ehttpc_pool:pick_worker(?POOL),
-                    exit(Worker, kill),
-                    ?assertMatch(
-                        {error, {ehttpc_worker_down, _}},
-                        ehttpc:health_check(Worker, 5_000)
-                    )
-                end
-            )
-        end},
-        {timeout, 20, fun() ->
-            ?WITH(
-                #{
-                    port => Port,
-                    name => ?FUNCTION_NAME,
-                    delay => 0
-                },
-                pool_opts(Unreachable, Port, true),
-                begin
-                    Worker = ehttpc_pool:pick_worker(?POOL),
-                    ?assertEqual(
-                        {error, connect_timeout},
-                        ehttpc:health_check(Worker, 5_000)
-                    )
-                end
-            )
-        end},
-        {timeout, 20, fun() ->
-            ?WITH(
-                #{
-                    port => Port,
-                    name => ?FUNCTION_NAME,
-                    delay => 0
-                },
-                pool_opts(Unknown, Port, true),
-                begin
-                    Worker = ehttpc_pool:pick_worker(?POOL),
-                    ?assertEqual(
-                        {error, nxdomain},
-                        ehttpc:health_check(Worker, 5_000)
-                    )
-                end
-            )
-        end},
-        {timeout, 20, fun() ->
```

Comment:
this test is removed because OTP 26 no longer checks timeout arg validity.
it crashes in inet module.